src/main/java/com/mathworks/ci/actions/RunMatlabBuildAction.java

```diff
@@ -93,32 +62,17 @@ public void run() throws IOException, InterruptedException, MatlabExecutionExcep
         } finally {
             annotator.forceEol();
 
-            try {
-                // Handle build result
-                Run<?,?> build = this.params.getBuild();
-                FilePath jsonFile = new FilePath(runner.getTempFolder(), "buildArtifact.json");
-                if (jsonFile.exists()) {
-                    FilePath rootLocation = new FilePath(
-                            new File(
-                                build.getRootDir().getAbsolutePath(),
-                                "buildArtifact" + this.getActionID() + ".json")
-                            );
-                    jsonFile.copyTo(rootLocation);
-                    jsonFile.delete();
-                    build.addAction(new BuildArtifactAction(build, this.getActionID()));
-                }
-            } catch (Exception e) {
-                // Don't want to override more important error
-                // thrown in catch block
-                System.err.println(e.toString());
-            } finally {
-                try {
-                    this.runner.removeTempFolder();
-                } catch (Exception e) {
-                    System.err.println(e.toString());
-                }
-            }
+            teardownAction();
         }
+    }
 
+    @Override
+    public void teardownAction() {
+        Run<?, ?> build = this.params.getBuild();
+        
+        // Handle build result
+        super.moveJsonArtifactToBuildRoot(build, MatlabBuilderConstants.BUILD_ARTIFACT);
+        
+        super.teardownAction();
     }
```

Comment:
Definitely not a blocking piece of feedback here, just something I'm wondering about.

Do we want to override this here, or could we change the parent `teardownAction` to check if it has an annotator and then move the artifact if so? Overrides are generally not my preference, because it can lead to a pattern of changes having to be done in each child object which overrides a method.

Maybe something like this?

MatlabAction.java
```java
public void teardownAction() {
    if (this.annotator != null) {
        this.moveJsonArtifactToBuildRoot(...)
    }

    try {
        this.runner.removeTempFolder();
    } catch (Exception e) {
        System.err.println(e.toString());
    }
}
```

Reply:
Even I wasn't completely convinced with this design but thought it made some sense here. Checking `null` for annotator could also be fine for now (atleast until we start using annotator somewhere in **Run MATLAB Tests** as well). If this feels better, I can update it.

@nbhoski do you have a preference here?